presto-tests/src/test/java/io/prestosql/execution/TestCoordinatorDynamicFiltering.java

```diff
@@ -160,6 +165,23 @@ public void testBroadcastJoinWithSelectiveBuildSide()
                         singleValue(BIGINT, 1L))));
     }
 
+    @Test(timeOut = 30_000)
+    public void testJoinWithImplicitCoercion()
+    {
+        // setup fact table with integer suppkey
+        computeActual("CREATE TABLE lineitem_decimal AS SELECT orderkey, CAST(suppkey as decimal(19, 0)) suppkey_decimal FROM tpch.tiny.lineitem");
+
+        assertQueryDynamicFilters(
+                "SELECT * FROM lineitem_decimal l JOIN tpch.tiny.supplier s ON l.suppkey_decimal = s.suppkey AND s.name >= 'Supplier#000000080'",
```

Comment:
it should be the other way around, `l.suppkey` (probe) should be joined with `s.suppkey_decimal` (build)

Reply:
right, modified this test now